pkg/processor/k8sprocessor/pod_association.go

```diff
@@ -32,9 +32,16 @@ import (
 func extractPodID(ctx context.Context, attrs pdata.AttributeMap, associations []kube.Association) (podIdentifierKey string, podIdentifierValue kube.PodIdentifier) {
 	hostname := stringAttributeFromMap(attrs, conventions.AttributeHostName)
 	var connectionIP kube.PodIdentifier
-	if c, ok := client.FromContext(ctx); ok {
-		connectionIP = kube.PodIdentifier(c.IP)
+
+	if c := client.FromContext(ctx); c.Addr != nil {
+		ip, _, err := net.SplitHostPort(c.Addr.String())
+		if err == nil {
+			connectionIP = kube.PodIdentifier(ip)
+		} else if ip := net.ParseIP(c.Addr.String()); ip != nil {
```

Comment:
My understanding is this is the most common path. How much extra processing time does it add? It will executed for each resourceX package